spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/http/WebClientEurekaHttpClient.java

```diff
@@ -33,108 +31,86 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.client.RestTemplate;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
 
 import static com.netflix.discovery.shared.transport.EurekaHttpResponse.anEurekaHttpResponse;
 
 /**
  * @author Daniel Lavoie
+ * @author Haytham Mohamed
  */
-public class RestTemplateEurekaHttpClient implements EurekaHttpClient {
+public class WebClientEurekaHttpClient implements EurekaHttpClient {
 
 	protected final Log logger = LogFactory.getLog(getClass());
 
-	private RestTemplate restTemplate;
+	private WebClient webClient;
 
-	private String serviceUrl;
```

Comment:
What happened to serviceUrl?

Reply:
the factory bakes it all and sets it as the base url on the WebClient builder